rosette/src/Ob.cc

```diff
@@ -275,6 +275,16 @@ int Ob::size() {
 }
 
 
+bool Ob::match_pattern(Ob* other) {
+    PROTECT_THIS(Ob);
+    if (IS_A(SELF, RBLstring)) {
+        return ((RBLstring*) SELF)->match_pattern(other);
+    } else {
+        return SELF == other;
+    }
+}
+
+
```

Comment:
This is a special case for strings. Do we want to allow other non-referential equalities besides strings?